### ISSUE TITLE
Not only @extend classes or placeholders but also @include them 

### ISSUE THREAD

**Issue by timfayz** [sass/sass#1284]
Again and again I (I think not only I) bump on the need to include properties from other classes (like in LESS). Let's take example:

``` SCSS
.shadow-out {
  box-shadow: ...pretty shadow...; 
}
%color-gray {
  color: ...gray...; 
}

textarea,
input:not([type]),
input[type="date"],
input[type="datetime"],
input[type="datetime-local"],
input[type="email"],
input[type="file"],
input[type="month"],
input[type="number"],
input[type="password"],
input[type="search"],
input[type="tel"],
input[type="text"],
input[type="time"],
input[type="url"],
input[type="week"] {
  @include .shadow-out;
  @include %color-gray;
  border: ...;
  //etc
}
```

Because mixins' names can't be started with . or % it make sense to parse this names differently.. Can this feature be implemented? From my point of view it is quite logical if we can do @extend, then we can do @include.
Like @extend this feature should work with dynamically generated classes and placeholders.

_For heaven's sake don't offer solutions like "you can do this...[your code], you can do that...[your code]"_

**Comment by nex3** [comment]
**tl;dr: Use `@extend`.**

What you're proposing is an ill-defined feature. First, it doesn't make sense to define this feature only for certain selector types; there's nothing special about class selectors or placeholders that makes this appropriate for them and not for, say, ids or element selectors. You'd need to make `@include` work for any selector (we'll ignore the syntactic ambiguity this causes with element selectors).

When you write `@include <selector>`, what does that mean? Does it mean "include all properties that were defined in CSS rules whose selector is exactly equal to <selector>"? To do so goes against the grain of CSS; it treats selectors like opaque strings of text, rather than semantic entities describing sets of HTML elements. `@include .foo.bar` shouldn't do something different than `@include .bar.foo`. Moreover, `@include .foo` shouldn't just ignore the styles defined in `.parent .foo { ... }`.

So how do we make it work _with_ the semantics of CSS, rather than _against_ them? We need to find a way of generalizing what you're trying to express when you write `@include .shadow-out`. What you want is for the styles that apply to an element with `class="shadow-out"` to apply to your textareas and inputs as well. In general, then, if you write `@include <selector>` you want all styles that apply to that selector to apply to the extending selector as well.

This is precisely the semantics of `@extend`. The implementation is different than a mixin include, but that shouldn't worry you; the styling works the same.
